composables/useChainRedirect.ts

```diff
@@ -0,0 +1,181 @@
+  .map<PageType>((value) => getEnumKeyByValue(PageType, value) as PageType)
+  .filter(Boolean) as PageType[]
+
+const pageAvailabilityPerChain = {
+  [PageType.PREFIX_EXPLORE_ITEMS]: (chain: Prefix): boolean =>
+    explorerVisible(chain),
+  [PageType.PREFIX_EXPLORE_COLLECTIBLES]: (chain: Prefix): boolean =>
+    explorerVisible(chain),
+  [PageType.SERIES_INSIGHT]: (chain: Prefix) => seriesInsightVisible(chain),
+  [PageType.PREFIX_CLASSIC_CREATE]: (chain: Prefix) => createVisible(chain),
+  [PageType.PREFIX_MASSMINT]: (chain: Prefix) => massmintCreateVisible(chain),
+  [PageType.PREFIX_MASSMINT_ONBOARDING]: (chain: Prefix) =>
+    massmintCreateVisible(chain),
+  [PageType.SALES]: (chain: Prefix) => salesVisible(chain),
+  [PageType.HOT]: (chain: Prefix) => hotVisible(chain),
+  [PageType.BLOG]: () => true,
+}
+
+const generateRouteRegexPattern = (pattern: string): string => {
+  const patternWithPlaceholderReplaced = pattern.replace(/\{.*\}/, '.+')
+  const patternWithHyphensEscaped = patternWithPlaceholderReplaced.replace(
+    '-',
+    '\\-'
+  )
+  return `^${patternWithHyphensEscaped}$`
+}
+
+const getPageType = (routeName: string): PageType => {
+  const matchingKey = Object.keys(PageType).find((key) => {
+    const pagePattern = PageType[key]
+    const regexPattern = generateRouteRegexPattern(pagePattern)
+    return new RegExp(regexPattern).test(routeName)
+  })
+
+  return matchingKey as PageType
+}
+
+export default function (allowRedirectIfCheckNotPresent = false) {
+  const route = useRoute()
+
+  const getChangedChainPrefixFromPath = (
+    chain: Prefix,
+    prevChain: Prefix
+  ): RedirectPath => ({
+    path: route.path.replace(prevChain, chain),
+    query: route.query,
+  })
+
+  const checkIfPageHasSpecialRedirect = (pageType: PageType): boolean => {
+    return SpecialRedirectPageTypes.includes(pageType as PageType)
+  }
+
+  const getPageRedirectPath = (
+    newChain: Prefix,
+    prevChain: Prefix,
+    defaultRedirectPath: string
+  ): RedirectPath | null => {
+    const routeName = route.name || ''
+
+    const defaultRedirect: RedirectPath = {
+      path: defaultRedirectPath,
+    }
+
+    const pageType = getPageType(routeName)
+
+    const hasSpecialRedirect = checkIfPageHasSpecialRedirect(pageType)
+
+    if (!hasSpecialRedirect) {
+      return defaultRedirect
+    }
+
+    const pageTypeValue = PageType[pageType]
+    const pageRedirectType = PageRedirectType[pageTypeValue]
+    const isStayRedirect = pageRedirectType === RedirectTypes.STAY
+
+    let isPageAvailableForChain = allowRedirectIfCheckNotPresent
+    const pageAvailabilityCheck = pageAvailabilityPerChain[pageTypeValue]
+
+    if (pageAvailabilityCheck) {
+      isPageAvailableForChain = pageAvailabilityCheck(newChain)
+    }
+
+    if (isStayRedirect && isPageAvailableForChain) {
+      return null
+    }
+
+    if (!isPageAvailableForChain || isStayRedirect) {
+      return defaultRedirect
+    }
```

Comment:
It would be easier to read.
```suggestion
    if (isStayRedirect) {
       if (isPageAvailableForChain) {
           return null
       } else {
           return defaultRedirect
       }
    }
   
```